File: backend/internal/domain/idea.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// IdeaStatus represents the lifecycle state of an idea.
type IdeaStatus string

const (
	IdeaStatusDraft      IdeaStatus = "draft"
	IdeaStatusValidating IdeaStatus = "validating"
	IdeaStatusValidated  IdeaStatus = "validated"
	IdeaStatusPromoted   IdeaStatus = "promoted"
	IdeaStatusArchived   IdeaStatus = "archived"
)

// PlatformType represents a social/content platform.
type PlatformType string

const (
	PlatformYouTube   PlatformType = "youtube"
	PlatformTikTok    PlatformType = "tiktok"
	PlatformInstagram PlatformType = "instagram"
	PlatformTwitter   PlatformType = "twitter"
	PlatformLinkedIn  PlatformType = "linkedin"
)

// Idea represents a content idea in its lifecycle.
type Idea struct {
	ID                  uuid.UUID            `json:"id"`
	WorkspaceID         uuid.UUID            `json:"workspace_id"`
	CreatedBy           uuid.UUID            `json:"created_by"`
	Title               string               `json:"title"`
	Description         *string              `json:"description,omitempty"`
	Status              IdeaStatus           `json:"status"`
	PlatformTarget      *PlatformType        `json:"platform_target,omitempty"`
	PromotedToContentID *uuid.UUID           `json:"promoted_to_content_id,omitempty"`
	Metadata            map[string]any       `json:"metadata,omitempty"`
	Tags                []string             `json:"tags"`
	ValidationScore     *IdeaValidationScore `json:"validation_score,omitempty"`
	CreatedAt           time.Time            `json:"created_at"`
	UpdatedAt           time.Time            `json:"updated_at"`
	DeletedAt           *time.Time           `json:"deleted_at,omitempty"`
}

// IdeaValidationScore holds AI-generated scores for an idea.
type IdeaValidationScore struct {
	NoveltyScore     int     `json:"novelty_score"`
	AudienceFitScore int     `json:"audience_fit_score"`
	ViabilityScore   int     `json:"viability_score"`
	UrgencyScore     int     `json:"urgency_score"`
	PersonalFitScore int     `json:"personal_fit_score"`
	OverallScore     int     `json:"overall_score"`
	AIReasoning      *string `json:"ai_reasoning,omitempty"`
}

// IdeaFilter holds optional filtering parameters for listing ideas.
type IdeaFilter struct {
	Status         *IdeaStatus
	CreatedBy      *uuid.UUID
	PlatformTarget *PlatformType
	Limit          int
	Offset         int
}

File: frontend/packages/types/src/idea.ts
import type { UUID, Timestamp } from './common';
import type { PlatformType } from './content';

export type IdeaStatus = 'draft' | 'validating' | 'validated' | 'promoted' | 'archived';

/** @deprecated Use IdeaStatus instead */
export type IdeaStage = 'raw' | 'refined' | 'scripted' | 'recorded' | 'published';

export interface IdeaValidationScore {
  novelty_score: number;
  audience_fit_score: number;
  viability_score: number;
  urgency_score: number;
  personal_fit_score: number;
  overall_score: number;
  ai_reasoning?: string | null;
}

export interface Idea {
  id: UUID;
  workspace_id: UUID;
  created_by: UUID;
  title: string;
  description?: string | null;
  status: IdeaStatus;
  platform_target?: PlatformType | null;
  promoted_to_content_id?: UUID | null;
  metadata?: Record<string, unknown> | null;
  tags: string[];
  validation_score?: IdeaValidationScore | null;
  created_at: Timestamp;
  updated_at: Timestamp;
  deleted_at?: Timestamp | null;

  // Legacy aliases kept for backward compatibility
  /** @deprecated Use created_by instead */
  user_id?: UUID;
  /** @deprecated Use IdeaStatus values instead */
  stage?: IdeaStage;
  /** @deprecated Use validation_score.overall_score instead */
  ai_summary?: string | null;
}
